internal/core/src/segcore/ChunkedSegmentSealedImpl.cpp

```diff
@@ -595,15 +601,36 @@ ChunkedSegmentSealedImpl::chunk_view_by_offsets(
 PinWrapper<const index::IndexBase*>
 ChunkedSegmentSealedImpl::chunk_index_impl(FieldId field_id,
                                            int64_t chunk_id) const {
+    std::shared_lock lck(mutex_);
```

Comment:
why adding this lock? I think this class will not receive read requests before it is fully loaded.